superset/views/core.py

```diff
@@ -2503,6 +2504,61 @@ def stop_query(self):
             pass
         return self.json_response('OK')
 
+    @has_access_api
+    @expose('/validate_sql_json/', methods=['POST', 'GET'])
+    @log_this
+    def validate_sql_json(self):
+        """Validates that arbitrary sql is acceptable for the given database.
+        Returns a list of error/warning annotations as json.
+        """
+        sql = request.form.get('sql')
+        database_id = request.form.get('database_id')
+        schema = request.form.get('schema') or None
+        template_params = json.loads(
+            request.form.get('templateParams') or '{}')
+
+        if len(template_params) > 0:
+            # TODO: factor the Database object out of template rendering
+            #       or provide it as mydb so we can render template params
+            #       without having to also persist a Query ORM object.
+            return json_error_response(
+                'SQL validation does not support template parameters')
+
+        session = db.session()
+        mydb = session.query(models.Database).filter_by(id=database_id).first()
+        if not mydb:
+            json_error_response(
+                'Database with id {} is missing.'.format(database_id))
+
+        spec = mydb.db_engine_spec
+        if not spec.engine in SQL_VALIDATORS_BY_ENGINE:
+            return json_error_response(
```

Comment:
Similarly, **most** databases are going to turn up empty here in at least the near-medium term. This should not return a 500, it is a reasonable, valid response to say "can't do this".

Reply:
Actually, the more I think about it, this might be a straight-up 200 response. Same with the one re templates. Thoughts?

Reply:
IMO 400 bad request is the right return — we're bombing out because the user asked us to do something we explicitly don't do, so we should let them know that.

This is assuming we're able to plumb through the "can I validate on this backend" db property to the client, of course.

Reply:
Yeah, makes sense.  I could see a case for either, but I think 400 is more to the point... 200 would imply we were combining "do you support this feature?" with "is this sql valid?" which would not be a clean API so... I agree with you.